components/leave_team_modal/leave_team_modal.jsx

```diff
@@ -4,79 +4,79 @@
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage, injectIntl, intlShape} from 'react-intl';
+import PropTypes from 'prop-types';
 
-import * as GlobalActions from 'actions/global_actions.jsx';
-import ModalStore from 'stores/modal_store.jsx';
-import UserStore from 'stores/user_store.jsx';
 import WebrtcStore from 'stores/webrtc_store.jsx';
-import {ActionTypes, WebrtcActionTypes} from 'utils/constants.jsx';
+import {WebrtcActionTypes} from 'utils/constants.jsx';
 
-class LeaveTeamModal extends React.Component {
-    constructor(props) {
-        super(props);
+class LeaveTeamModal extends React.PureComponent {
+    static propTypes = {
 
-        this.handleToggle = this.handleToggle.bind(this);
-        this.handleSubmit = this.handleSubmit.bind(this);
-        this.handleHide = this.handleHide.bind(this);
-        this.handleKeyPress = this.handleKeyPress.bind(this);
+        /**
+         * Current user.
+         */
+        currentUser: PropTypes.object,
 
-        this.state = {
-            show: false,
-        };
-    }
+        /**
+         * hide action
+         */
 
-    componentDidMount() {
-        ModalStore.addModalListener(ActionTypes.TOGGLE_LEAVE_TEAM_MODAL, this.handleToggle);
-        document.addEventListener('keypress', this.handleKeyPress);
-    }
+        onHide: PropTypes.func.isRequired,
 
-    componentWillUnmount() {
-        ModalStore.removeModalListener(ActionTypes.TOGGLE_LEAVE_TEAM_MODAL, this.handleToggle);
-        document.removeEventListener('keypress', this.handleKeyPress);
-    }
+        /**
+         * show or hide modal
+         */
 
-    handleKeyPress(e) {
-        if (e.key === 'Enter' && this.state.show) {
-            this.handleSubmit(e);
-        }
-    }
+        show: PropTypes.bool.isRequired,
+
+        /**
+         * is the user busy in a video call
+         */
+
+        isBusy: PropTypes.bool.isRequired,
+
+        intl: intlShape.isRequired,
+
+        actions: PropTypes.shape({
 
-    handleToggle(value) {
-        this.setState({
-            show: value,
-        });
+            /**
+             * An action to remove user from team
+             */
+
+            removeUserFromTeam: PropTypes.func.isRequired,
+
+            /**
+             * An action to toggle the right menu
+             */
+
+            toggleSideBarRightMenu: PropTypes.func.isRequired,
+        }),
+    };
+
+    constructor(props) {
+        super(props);
+        this.handleSubmit = this.handleSubmit.bind(this);
     }
 
     handleSubmit(e) {
-        this.setState({
-            show: false,
-        });
+        this.props.onHide();
 
-        if (WebrtcStore.isBusy()) {
+        if (this.props.isBusy) {
             WebrtcStore.emitChanged({action: WebrtcActionTypes.IN_PROGRESS});
```

Comment:
I have a question regarding this. Even if we convert this to redux action , I assume somewhere down the chain we need to emit 
```
WebrtcStore.emitChanged({action: WebrtcActionTypes.IN_PROGRESS});
```
so that the webrtcStore can act on it.

Where will that happen ? Kindly clarify, if I am getting it wrong .

Reply:
Like on our offline DM, webrtcstore conversion to redux store is good to have in a separate PR.  I'm good with not changing `WebrtcStore.emitChanged`.